request17.go

```diff
@@ -67,6 +70,9 @@ func (r *Request) Context() context.Context {
 // documentation.
 func (r *Request) SetContext(ctx context.Context) *Request {
 	r.ctx = ctx
+	if r.RawRequest != nil {
```

Comment:
@clippit I have intentionally did not make these changes. Because by resty design PreRequestHook is only meant to modify `RawRequest`  not resty Request` (refer to #62 ). Any changes made in resty Request instance will not affect the request.